### DISCUSSION TITLE
Dolby Vision profile 5 rpu injection to a hdr10 video

### DISCUSSION THREAD

**Discussion by draco1544** [quietvoid/dovi_tool#190] [General]
Hello,
I have a Dolby Vision profile 5 rpu, it's possible to inject in a classic hdr10 video file ? I want to replace the HDR 10

**Comment by quietvoid** [comment]
Edit the RPU with mode 3 and inject it.
Example JSON: https://github.com/quietvoid/dovi_tool/blob/main/assets/editor_examples/mode.json

**Reply by draco1544** [reply]
> Edit the RPU with mode 3 and inject it. Example JSON: https://github.com/quietvoid/dovi_tool/blob/main/assets/editor_examples/mode.json

The output file will be DV profile 5 ?

**Reply by quietvoid** [reply]
> The output file will be DV profile 5 ?

No, it will be profile 8.1, which is what you want for an HDR10 base video.

**Reply by draco1544** [reply]
> > The output file will be DV profile 5 ?
> 
> 
> 
> No, it will be profile 8.1, which is what you want for an HDR10 base video.

It's not possible to remove the hdr 10 then inject the profile 5 RPU ? I want a DV profile 5 output 😂

**Reply by quietvoid** [reply]
> It's not possible to remove the hdr 10 then inject the profile 5 RPU ? I want a DV profile 5 output joy

Then no, that's not possible with free tools.

**Reply by draco1544** [reply]
The Apple TV is unfortunately not compatible with DV profile 8.1, so I need to apply a profile 5 rpu to a hdr 10 based video (I want the final file to be DV only)

**Reply by quietvoid** [reply]
The only workaround is to modify the mp4 for Apple devices.
There's this guide about it: https://www.avsforum.com/threads/apple-dolby-vision.3252865/#post-61936696

Specifically the `Hybrid Profile 5 and HDR10 Blu-ray to get unofficial MEL` section.
There is also this project: https://github.com/nilaoda/DoViMuxer

**Reply by draco1544** [reply]
> The only workaround is to modify the mp4 for Apple devices.
> 
> There's this guide about it: https://www.avsforum.com/threads/apple-dolby-vision.3252865/#post-61936696
> 
> 
> 
> Specifically the `Hybrid Profile 5 and HDR10 Blu-ray to get unofficial MEL` section.
> 
> There is also this project: https://github.com/nilaoda/DoViMuxer

Thank a lot ! Just a last thing : 

I have a mkv file with this meta data : `HDR format : Dolby Vision, Version 1.0, dvhe.05.06, BL+RPU / SMPTE ST 2086, HDR10 compatible / Codec ID : V_MPEGH/ISO/HEVC`

The Dolby vision is triggered on the Apple TV. With this file I will get correct Dolby vision ? I have extracted a rpu from a dv profile 5 mkv then I injected this rpu to a hdr 10 mkv file.

**Reply by quietvoid** [reply]
No, if you use Infuse for MKV playback, it will not be correct.
I don't know what players support the mp4 workaround.

Maybe @sw5163 can help you.

**Reply by sw5163** [reply]
Any player that passes DV content to Apple's native player core should play it.
For example, Plex, Safari, File App, QuickTime.
The file must specifically be tagged as P5 dvh1.

**Reply by draco1544** [reply]
> Any player that passes DV content to Apple's native player core should play it.
> 
> For example, Plex, Safari, File App, QuickTime.
> 
> The file must specifically be tagged as P5 dvh1.

Infuse will display true Dolby vision with an MP4 Fake P5 dvh1 file ?

**Reply by sw5163** [reply]
I don't think so.
I would recommend first converting quietvoid's test pattern as a practice to see if you workflow is correct.
His link seems to be expired.
And you could try these already converted files on your player to see if DV plays correctly.
https://drive.google.com/drive/folders/1IZXfZKL9AuZ2FRViysGzo3CJdcEx3Cjs
More tests:
https://drive.google.com/drive/folders/1FwwiVY7hUpWt1OmMpLguALPs8NfU53Ee

**Reply by draco1544** [reply]
> I don't think so.
> 
> I would recommend first converting quietvoid's test pattern as a practice to see if you workflow is correct.
> 
> His link seems to be expired.
> 
> And you could try these already converted files on your player to see if DV plays correctly.
> 
> https://drive.google.com/drive/folders/1IZXfZKL9AuZ2FRViysGzo3CJdcEx3Cjs
> 
> More tests:
> 
> https://drive.google.com/drive/folders/1FwwiVY7hUpWt1OmMpLguALPs8NfU53Ee

Is the Dolby vision profile 8.1 playable on most players (except Apple TV and iOS devices)?

**Reply by sw5163** [reply]
> Is the Dolby vision profile 8.1 playable on most players (except Apple TV and iOS devices)?

On supported DV devices, yes.
RESET_9999 provides a very useful spreadsheet for DV devices support.
https://docs.google.com/spreadsheets/d/1jBIGF8XTVi9VmDBZ8a5hEyongYMCDlUiLHU9n1f_S74/edit#gid=427220017
Under his signature [Dolby Vision Stuff]
https://forum.makemkv.com/forum/viewtopic.php?p=123895#p123895

**Reply by draco1544** [reply]
> > Is the Dolby vision profile 8.1 playable on most players (except Apple TV and iOS devices)?
> 
> 
> 
> On supported DV devices, yes.
> 
> RESET_9999 provides a very useful spreadsheet for DV devices support.
> 
> https://docs.google.com/spreadsheets/d/1jBIGF8XTVi9VmDBZ8a5hEyongYMCDlUiLHU9n1f_S74/edit#gid=427220017
> 
> Under his signature [Dolby Vision Stuff]
> 
> https://forum.makemkv.com/forum/viewtopic.php?p=123895#p123895

I use his Dolby vision script, Fake p5 test work with plex app but not with infuse, do you have an idea how to make fake p5 for infuse ?

**Reply by sw5163** [reply]
As far as I know, some older version of Infuse passes fake P5 to Apple's player core, so it can play DV properly.
However, later version of Infuse detects DV profile and they think P5 with HDR10 BL is not correct.
So it plays as normal HDR.
I don't think there is any way for Infuse to play P8.1 for now until its developers change it.
